### PR TITLE
add tests for user log out

### PR DESCRIPTION
**Pull request by lestairon**
## Description:
Allow User to sign out of the app.

P.S:
This functionality is already implemented by devise-token-auth. The meaning of this PR is to make sure it is working with our current configuration.

### CODE REVIEW

File: spec/requests/api/v1/users/sign_out_spec.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+      expect(response).to have_http_status(:not_found)
+      expect(response.body).to eq({ success: false,
+                                    errors: ['User was not found or was not logged in.'] }.to_json)
+    end
+  end
+
+  context 'when user tokens are incorrect' do
+    let(:user_tokens) { user.create_new_auth_token }
+    before do
```
Comment (Ricoch) [inline spec/requests/api/v1/users/sign_out_spec.rb]: Add an empty line between these two (https://github.com/rubocop/rspec-style-guide#let-grouping)

File: spec/requests/api/v1/users/sign_out_spec.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
+require 'rails_helper'
+
+RSpec.describe 'User sign out', type: :request do
+  let(:user) { create(:user) }
+  subject { delete destroy_api_v1_user_session_path, headers: user_tokens, as: :json }
+
+  context 'when user tokens are correct' do
+    before { subject }
+    let(:user_tokens) { user.create_new_auth_token }
```
Comment (Ricoch) [inline spec/requests/api/v1/users/sign_out_spec.rb]: Same here about empty line

File: spec/requests/api/v1/users/sign_out_spec.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
+require 'rails_helper'
+
+RSpec.describe 'User sign out', type: :request do
+  let(:user) { create(:user) }
+  subject { delete destroy_api_v1_user_session_path, headers: user_tokens, as: :json }
```
Comment (Ricoch) [inline spec/requests/api/v1/users/sign_out_spec.rb]: Same here https://github.com/rubocop/rspec-style-guide#empty-line-after-let (event though it is not clear in the example, the description says it)

File: spec/requests/api/v1/users/sign_out_spec.rb [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+      expect(response.body).to eq({ success: true }.to_json)
+    end
+
+    it 'returns errors when signin out again' do
+      delete destroy_api_v1_user_session_path, headers: user_tokens, as: :json
+
+      expect(response).to have_http_status(:not_found)
+      expect(response.body).to eq({ success: false,
+                                    errors: ['User was not found or was not logged in.'] }.to_json)
+    end
```
Comment (Ricoch) [inline spec/requests/api/v1/users/sign_out_spec.rb]: WDYT of doing it like this:
```suggestion
    context 'when the user is already signed out' do
      before do
        delete destroy_api_v1_user_session_path, headers: user_tokens, as: :json
      end
      
      it 'returns not found' do
        expect(response).to have_http_status(:not_found)
      end
      
      it 'returns the corresponding error' do
        expect(response.body).to eq({ success: false,
                                      errors: ['User was not found or was not logged in.'] }.to_json)
      end
    end
```
?

File: spec/requests/api/v1/users/sign_out_spec.rb [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  end
+
+  context 'when user tokens are incorrect' do
+    let(:user_tokens) { user.create_new_auth_token }
+    before do
+      user_tokens.merge!('access-token': 'incorrect_token')
+      subject
```
Comment (Ricoch) [inline spec/requests/api/v1/users/sign_out_spec.rb:44]: Not sure if there is a rule for this one, but I personally like to add an empty line in between these as well, but I'll leave that one up to your preferred choice

Comment (lestairon) [reply spec/requests/api/v1/users/sign_out_spec.rb:44]: I think in this case it's good having them together, in blocks when there's an empty space before the last line, at least to me means that the last line will be the returned value of said block.

**Review by mrodriguez90 — APPROVED** [review]

**Review by Ricoch — APPROVED** [review]

**Pull request merged**